docs/developer-docs/multi-chain/generating-addresses.mdx

```diff
@@ -0,0 +1,264 @@
+
+:::
+
+<AdornedTabs groupId="chain">
+<TabItem value="bitcoin" label="Bitcoin" default>
+
+A Bitcoin address is a hash value. There are currently two types of Bitcoin addresses: P2SH and P2PKH. An address's full public key cannot be provided when sending transactions, so to redeem transactions sent to your Bitcoin address, you must provide both the transaction signature and the public key used to sign the transaction.
+
+To submit and receive Bitcoin transactions in a canister, you need to generate a Bitcoin address for that canister.
+
+Currently, the `ecdsa_public_key` API supports generating P2PKH addresses.
+
+#### Generating a P2PKH address
+
+To generate a P2PKH address for a canister, make a call to the `ecdsa_public_key_api`:
+
+<AdornedTabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+
+/// Returns the P2PKH address of this canister at a specific derivation path.
+    public func get_p2pkh_address() : async BitcoinAddress {
+        await BitcoinWallet.get_p2pkh_address(NETWORK, KEY_NAME, DERIVATION_PATH)
+    };
+
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+
+/// Returns the P2PKH address of this canister at a specific derivation path.
+#[update]
+pub async fn get_p2pkh_address() -> String {
+    let derivation_path = DERIVATION_PATH.with(|d| d.clone());
+    let key_name = KEY_NAME.with(|kn| kn.borrow().to_string());
+    let network = NETWORK.with(|n| n.get());
+    bitcoin_wallet::get_p2pkh_address(network, key_name, derivation_path).await
+}
+
+```
+</TabItem>
+</AdornedTabs>
+
+[Learn more about Bitcoin addresses](https://en.bitcoin.it/wiki/Transaction#Pay-to-PubkeyHash).
+
+[Learn more about the `ecdsa_public_key` API](/docs/current/references/ic-interface-spec/#ic-ecdsa_public_key).
+
+#### Bitcoin addresses associated with Internet Identity
+
+By default, your Internet Identity will have a Bitcoin address that can be used to send and receive BTC. You can view this address by logging into the [NNS frontend dapp](https://nns.ic0.app/) with your [Internet Identity](/docs/current/developer-docs/web-apps/user-login/internet-identity/overview).
+
+</TabItem>
+
+<TabItem value="ethereum" label="Ethereum & EVM">
+
+On Ethereum and other EVM chains, there are two types of addresses: wallet addresses for holding assets and contract addresses. Wallet addresses are public accounts that can send and receive ETH tokens. Contract addresses refer to a smart contract that has been executed on Ethereum and the transactions associated with that contract.
+
+:::caution
+
+Not sure if a chain is EVM-compatible? Check [ChainList](https://chainlist.org/), a full database of EVM-compatible chains.
+
+:::
+
+Wallet and contract addresses can be queried using the [EVM RPC canister](/docs/current/developer-docs/multi-chain/ethereum/evm-rpc/evm-rpc-canister).
+
+#### Generating an ETH address
```

Comment:
Should we note "Generating an ETH or EVM address"?

Reply:
Revised